core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.util.TextFormatter;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+import io.netty.channel.EventLoopGroup;
+
+/**
+ * A {@link ServerErrorHandler} that wraps another {@link ServerErrorHandler}
+ * to periodically report the exceptions that were not logged by decorators such as
+ * {@link LoggingService}.
+ */
+class ExceptionReportingServerErrorHandler implements ServerErrorHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ExceptionReportingServerErrorHandler.class);
+    private final ServerErrorHandler delegate;
+    private final LongAdder counter;
+    private final Duration duration;
+    private boolean started;
+    @Nullable
+    private Throwable thrownException;
+    @Nullable
+    private ScheduledFuture<?> reportingTaskFuture;
+
+    ExceptionReportingServerErrorHandler(ServerErrorHandler serverErrorHandler, Duration duration) {
+        assert !duration.isNegative() && !duration.isZero();
+        delegate = serverErrorHandler;
+        this.duration = duration;
+        counter = new LongAdder();
```

Comment:
How about also recording this value to `meterRegistry` as a default server metric such as we do for connections and pending responses? 
https://github.com/line/armeria/blob/64e8fa91b45098a454e8e49ea13644bb2f465e8c/core/src/main/java/com/linecorp/armeria/server/Server.java#L554

Reply:
That's a great idea. Will it be ok to register gauge like below ? 
```
meterRegistry.gauge("armeria.server.unloggedExceptions", errorHandler.getNumberOfUnloggedExceptions()) ...
```

Reply:
That looks nice. 👍

Reply:
Shouldn't it be a counter rather than a gauge?

Reply:
Hmm, "counter" seems to be appropriate as well. But when I check the docs of `MeterRegistry`'s `counter` method
```
    /**
     * Tracks a monotonically increasing value.
     * @param name The base metric name
     * @param tags Sequence of dimensions for breaking down the name.
     * @return A new or existing counter.
     */
    public Counter counter(String name, Iterable<Tag> tags) {
        return Counter.builder(name).tags(tags).register(this);
    }
```
it says it `Tracks a monotonically increasing value`. But the unloggedExceptions counter is reset every fixed interval so... I wonder if it's an appropriate usage ?

Reply:
You can just keep incrementing the counter. It's the responsibility of the observability stack to track the increment. If we reset it periodically, it will have hard time figuring out the rate of exceptions.

Reply:
I changed `ExceptionReportingServerErrorHandler` to use `counter` instead, but I wonder it's the right way to use it. When I try to use default `MeterRegistry`'s `counter(String name, String.. tags)` it uses `io.micrometer.core.instrument.noop.NoopCounter` which doesn't do anything when `increment()` is called. So I used `MeterRegistry.More`'s `counter(String name, Iterable<Tag> tags, T obj, ToDoubleFunction<T> countFunction)` method and added `CumulativeCounter` to count unhandled exceptions. 


```java 
// Initialize counter in constructor 
counter = new CumulativeCounter(new Id("armeria.server.unhandledExceptions", Tags.empty(),                                          null, null, Type.COUNTER));

// ... 
// register on the meterRegistry using Meter.More's counter method 
server.config().meterRegistry().more().counter(counter.getId().getName(), counter.getId().getTags(), counter, CumulativeCounter::count);
```

Reply:
I'll use `meterRegistry.gauge` and `LongAdder` to collect metrics about unhandled exceptions.